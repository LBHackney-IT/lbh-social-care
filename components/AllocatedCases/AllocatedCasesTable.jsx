import Link from 'next/link';

const header = [
  'Person',
  'Name',
  'Address',
  'Date of birth',
  'Assessment type',
];

const AllocatedCasesTable = ({ cases }) => (
  <table className="govuk-table">
    <thead className="govuk-table__head">
      <tr className="govuk-table__row">
        {header.map((text) => (
          <th key={text} className="govuk-table__header">
            {text}
          </th>
        ))}
      </tr>
    </thead>
    <tbody className="govuk-table__body">
      {cases.map((cell) => (
        <tr key={cell.id} className="govuk-table__row">
          <td className="govuk-table__cell">{cell.personId} </td>
          <td className="govuk-table__cell">
            {
              <Link href={`/people/${cell.personId}`}>
                <a className="govuk-link">{cell.personName}</a>
              </Link>
            }
          </td>
          <td className="govuk-table__cell"> {cell.personAddress}</td>
          <td className="govuk-table__cell">
            {' '}
            {new Date(cell.personDateOfBirth).toLocaleDateString('en-GB')}
          </td>
          <td className="govuk-table__cell"> {cell.allocatedWorkerTeam}</td>
        </tr>
      ))}
    </tbody>
  </table>
);

export default AllocatedCasesTable;
