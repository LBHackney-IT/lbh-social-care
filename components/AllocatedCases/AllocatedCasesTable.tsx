import Link from 'next/link';

import { Allocation } from 'types';

const header = ['Person', 'Name', 'Address', 'Date of birth', 'Allocated Date'];

const AllocatedCasesTable = ({
  cases,
}: {
  cases: Allocation[];
}): React.ReactElement => (
  <table className="govuk-table">
    <thead className="govuk-table__head">
      <tr className="govuk-table__row">
        {header.map((text) => (
          <th key={text} className="govuk-table__header">
            {text}
          </th>
        ))}
      </tr>
    </thead>
    <tbody className="govuk-table__body">
      {cases.map((cell) => (
        <tr key={cell.id} className="govuk-table__row">
          <td className="govuk-table__cell">{cell.personId} </td>
          <td className="govuk-table__cell">
            {
              <Link href={`/people/${cell.personId}`}>
                <a className="govuk-link">{cell.personName}</a>
              </Link>
            }
          </td>
          <td className="govuk-table__cell"> {cell.personAddress}</td>
          <td className="govuk-table__cell">
            {' '}
            {new Date(cell.personDateOfBirth).toLocaleDateString('en-GB')}
          </td>
          <td className="govuk-table__cell">
            {' '}
            {new Date(cell.allocationStartDate).toLocaleDateString('en-GB')}
          </td>
        </tr>
      ))}
    </tbody>
  </table>
);

export default AllocatedCasesTable;
