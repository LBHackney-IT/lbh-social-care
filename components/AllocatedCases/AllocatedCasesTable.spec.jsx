import { render } from '@testing-library/react';

import AllocatedCasesTable from './AllocatedCasesTable';

describe('AllocatedCasesTable component', () => {
  const props = {
    cases: [
      {
        id: 123,
        personId: '87987',
        personName: 'James Smith',
        personAddress: 'Flat 11, Test road, E2 8TF',
        personDateOfBirth: '2021-01-21T09:38:02.186078',
        allocatedWorkerTeam: 'Long Term - case type: reviews',
      },
    ],
  };

  it('should render records properly', () => {
    const { asFragment } = render(<AllocatedCasesTable {...props} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
