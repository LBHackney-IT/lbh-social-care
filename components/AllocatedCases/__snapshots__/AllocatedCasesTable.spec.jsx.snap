// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AllocatedCasesTable component should render records properly 1`] = `
<DocumentFragment>
  <table
    class="govuk-table"
  >
    <thead
      class="govuk-table__head"
    >
      <tr
        class="govuk-table__row"
      >
        <th
          class="govuk-table__header"
        >
          Person
        </th>
        <th
          class="govuk-table__header"
        >
          Name
        </th>
        <th
          class="govuk-table__header"
        >
          Address
        </th>
        <th
          class="govuk-table__header"
        >
          Date of birth
        </th>
        <th
          class="govuk-table__header"
        >
          Assessment type
        </th>
      </tr>
    </thead>
    <tbody
      class="govuk-table__body"
    >
      <tr
        class="govuk-table__row"
      >
        <td
          class="govuk-table__cell"
        >
          87987 
        </td>
        <td
          class="govuk-table__cell"
        >
          <a
            class="govuk-link"
            href="/people/87987"
          >
            James Smith
          </a>
        </td>
        <td
          class="govuk-table__cell"
        >
           Flat 11, Test road, E2 8TF
        </td>
        <td
          class="govuk-table__cell"
        >
           1/21/2021
        </td>
        <td
          class="govuk-table__cell"
        >
           Long Term - case type: reviews
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;
